tests/integration/test_vulnerability_detector/test_general_settings/test_general_settings_ignore_time.py

```diff
@@ -28,72 +27,68 @@
 
 # fixtures
-
 @pytest.fixture(scope='module', params=configurations)
 def get_configuration(request):
     """Get configurations from the module."""
     return request.param
 
 
 #functions
-
 def generate_default_alert():
+    """
+    Generate an alert with the default values
+    """
     control_service('stop', daemon='wazuh-db')
     sleep(10)
-    clean_table(os.path.join(PACKAGES_DB_PATH,f"000.db"), 'sys_programs')
-    delete_vulnerability(DEFAULT_VULNERABILITY_ID)
+    clean_vd_tables()
     insert_package()
     insert_vulnerability()
-    modify_system()
-    control_service('start', daemon='wazuh-db')
-
-
-def reset_last_scan():
-    control_service('stop', daemon='wazuh-db')
     update_last_scan()
+    modify_system()
     control_service('start', daemon='wazuh-db')
 
 
 # Tests
-
-@pytest.mark.parametrize('tags_to_apply, custom_callback', [
-    ({'test_ignore_time'}, make_vuln_callback(callback_string))
-])
-def test_ignore_time(tags_to_apply, custom_callback, get_configuration, configure_environment, restart_modulesd):
+def test_ignore_time(get_configuration, configure_environment, restart_modulesd,
+                     tags_to_apply = 'test_ignore_time',
+                     custom_callback_vulnerability = make_vuln_callback(callback_string_vulnerability)):
     """
     Check if an alert is not fired during the ingnore time  interval
     """
-    reset_last_scan()
     generate_default_alert()
     ignore_time = get_configuration['metadata']['ignore_time']
-    seconds_to_travel = time_to_seconds(ignore_time)/2
+    jumps = get_configuration['metadata']['jumps']
+    seconds_to_travel = time_to_seconds(ignore_time)/jumps
 
+    # Check for initial alert
     wazuh_log_monitor.start(timeout=get_configuration['metadata']['timeout'],
-                        callback=custom_callback,
-                        error_message='Alert did not appear at the start of the test')
-
-    check_time_travel(time_travel=True, interval=timedelta(seconds=seconds_to_travel))
-
-    with pytest.raises(TimeoutError):
-        wazuh_log_monitor.start(timeout=get_configuration['metadata']['timeout'],
-                                callback=custom_callback)
-        raise AttributeError('Alert appeared before ignore_time was finished')
-
+                            callback=custom_callback_vulnerability,
+                            error_message='Alert did not appear at the start of the test')
+
+    # Check if alert does not appear during ignore time
+    for i in range(1, jumps):
+        check_time_travel(time_travel=True, interval=timedelta(seconds=seconds_to_travel))
+        with pytest.raises(TimeoutError):
+            wazuh_log_monitor.start(timeout=get_configuration['metadata']['timeout'],
+                                    callback=custom_callback_vulnerability)
+            raise AttributeError('Alert appeared before ignore_time was finished')
```

Comment:
In loops like this where you don't use the variable `i` you can just do:
```suggestion
    for _ in range(1, jumps):
        check_time_travel(time_travel=True, interval=timedelta(seconds=seconds_to_travel))
        with pytest.raises(TimeoutError):
            wazuh_log_monitor.start(timeout=get_configuration['metadata']['timeout'],
                                    callback=custom_callback_vulnerability)
            raise AttributeError('Alert appeared before ignore_time was finished')
```

Reply:
Done in 77223ed24d8fff611947e817d180bf744720ed4c